jerry-core/ecma/base/ecma-alloc.c

```diff
@@ -80,12 +80,37 @@ JERRY_STATIC_ASSERT (sizeof (ecma_extended_object_t) - sizeof (ecma_object_t) <=
   ALLOC (ecma_type) \
   DEALLOC (ecma_type)
 
-DECLARE_ROUTINES_FOR (object)
 DECLARE_ROUTINES_FOR (number)
 DECLARE_ROUTINES_FOR (collection_header)
 DECLARE_ROUTINES_FOR (collection_chunk)
-DECLARE_ROUTINES_FOR (string)
-DECLARE_ROUTINES_FOR (getter_setter_pointers)
+
+/**
+ * Allocate memory for ecma-object
+ *
+ * @return pointer to allocated memory
+ */
+inline ecma_object_t * __attr_always_inline___
+ecma_alloc_object (void)
```

Comment:
> I don't like macros because they are very hard to debug. I would prefer to keep the current code.

That can be a valid point. But here, we have a slight inconsistency now. Macro generated alloc routines have and assert on non-null allocation. Should all allocs have that (also the now-written routines) or is it not needed anymore and can be dropped from the remaining templates, too?